src/error.rs

```diff
@@ -0,0 +1,59 @@
+
+use std::{fmt::{self, Display, Debug}, ops::{Deref, DerefMut}};
+
+/// A generic error type.
+pub struct Error(anyhow::Error);
+
+impl Error {
+    /// Use this when you need to implement middleware, where the error type must be `surf::Error`.
+    pub fn new<E>(error: E) -> Self
```

Comment:
Let's make this constructor private or `pub(crate)`.

Reply:
I originally did this, but I found that the user needed to be able to create `surf::Errors` in the middleware examples. I'll have a look to see if this can be avoided.